src/song_builder.ts

```diff
@@ -189,8 +189,14 @@ class SongBuilder {
 
   startSection(sectionType: ParagraphType, tag: Tag): void {
     this.checkCurrentSectionType(NONE, tag);
-    this.sectionType = sectionType;
-    this.setCurrentProperties(sectionType);
+
+    if (sectionType === PART && tag.value) {
+      this.sectionType = tag.value.split(' ')[0].toLowerCase();
+    } else {
+      this.sectionType = sectionType;
```

Comment:
Maybe we should add a warning for a tag without value, and ignore it?

Reply:
This specifically handled the case where the opening is {sop: Tag} and the closing tag is {eop} where the sectionType is  the value and the incoming tag value is null. 

I feel like it would get perfectly acceptable to have a sop eop without values and it would just get processed as a sectionType 'part'

Thoughts?

Reply:
actually sorry I thought this comment was actually for this part

```
checkCurrentSectionType(sectionType: ParagraphType, tag: Tag): void {
    if (this.sectionType !== sectionType && !(sectionType === 'part' && tag.name === 'end_of_part')) {
      this.addWarning(`Unexpected tag {${tag.originalName}}, current section is: ${this.sectionType}`, tag);
    }
  }
```

ignore the first part of my comment. 

But I still think having sop without value should be valid with no warnings.